### PR TITLE
Pisp/feat/305 authorizations

### PR DESCRIPTION
**Pull request by eoln**
add new endpoint: POST /authorizations

### CODE REVIEW

**Comment by kleyow** [comment]
This merge should be pointing to `pisp/master`, yes?

File: src/domain/authorizations/authorizations.js [hunk 1]
```diff
@@ -88,43 +110,72 @@ const forwardAuthorizationMessage = async (headers, transactionRequestId, payloa
...
+      )
       await childSpan.error(fspiopError, state)
       await childSpan.finish(fspiopError.message, state)
     }
   }
 }
 
 /**
- * Forwards PUT authorization errors to destination FSP
+ * Forwards PUT/POST authorization errors to destination FSP
```
Comment (rmothilal) [inline src/domain/authorizations/authorizations.js:134]: i think this is only for the PUT

File: src/interface/swagger.json [hunk 2]
```diff
@@ -2114,52 +2250,60 @@
...
                 "ilpPacket",
                 "condition"
             ],
-            "type": "object",
             "properties": {
                 "transferAmount": {
-                    "$ref": "#/definitions/Money",
-                    "description": "The amount of money that the Payee FSP should receive."
+                    "description": "The amount of money that the Payee FSP should receive.",
+                    "type": "object",
```
Comment (rmothilal) [inline src/interface/swagger.json:2264]: i dont think type is required as the "Money" definition include type

File: src/interface/swagger.json [hunk 3]
```diff
@@ -2114,52 +2250,60 @@
...
+                    "$ref": "#/definitions/Money"
                 },
                 "payeeReceiveAmount": {
-                    "$ref": "#/definitions/Money",
-                    "description": "The amount of Money that the Payee should receive in the end-to-end transaction. Optional as the Payee FSP might not want to disclose any optional Payee fees."
+                    "description": "The amount of Money that the Payee should receive in the end-to-end transaction. Optional as the Payee FSP might not want to disclose any optional Payee fees.",
+                    "type": "object",
```
Comment (rmothilal) [inline src/interface/swagger.json:2269]: i dont think type is required as the "Money" definition include type

File: src/interface/swagger.json [hunk 4]
```diff
@@ -2114,52 +2250,60 @@
...
+                    "$ref": "#/definitions/Money"
                 },
                 "payeeFspFee": {
-                    "$ref": "#/definitions/Money",
-                    "description": "Payee FSP’s part of the transaction fee."
+                    "description": "Payee FSP’s part of the transaction fee.",
+                    "type": "object",
```
Comment (rmothilal) [inline src/interface/swagger.json:2274]: i dont think type is required as the "Money" definition include type

File: src/interface/swagger.json [hunk 5]
```diff
@@ -2114,52 +2250,60 @@
...
+                    "$ref": "#/definitions/Money"
                 },
                 "payeeFspCommission": {
-                    "$ref": "#/definitions/Money",
-                    "description": "Transaction commission from the Payee FSP."
+                    "description": "Transaction commission from the Payee FSP.",
+                    "type": "object",
```
Comment (rmothilal) [inline src/interface/swagger.json:2279]: i dont think type is required as the "Money" definition include type

**Comment by kleyow** [comment]
lgtm after rmothilal's comments are addressed.

File: src/domain/authorizations/authorizations.js (hunk 1 above)
Comment (eoln) [reply src/domain/authorizations/authorizations.js:134]: forwardAuthorizationMessage is also used to forward POST to PISP party so the auth errors for POST should also be handled

File: src/interface/swagger.json (hunk 2 above)
Comment (eoln) [reply src/interface/swagger.json:2264]: yeah, it isn't required, fixed

File: src/interface/swagger.json (hunk 4 above)
Comment (eoln) [reply src/interface/swagger.json]: fixed

**Pull request merged**